src/utility.c

```diff
@@ -571,21 +571,26 @@ int _count_args(char **args) {
  * Aborts on errors.
  */
 char **merge_args(char **args1, char **args2) {
-    int nArgs = 0;
+    size_t nArgs = 0;
     int s_index, d_index;
     char **newargs;
-    nArgs = _count_args(args1) + _count_args(args2) - 1;
+    nArgs = _count_args(args1) + _count_args(args2) + 1;
```

Comment:
_count_args includes the NULL terminator in the count.  So the -1 should be correct.  I think this will make an array that is 2 larger than needed.  Did coverity report this?

Reply:
coverity was mostly concerned about the user input being able to control the size of the buffer allocated; also that it was a signed type -- non-sensical to allocate negative memory.  I started by adding explicit limits in _count_args to ensure that the size of nArgs was bounded to much less than the limit of size_t (64 bit unsigned integer).

In any case, unsafe to have a -1 with an unsigned type needed to fix the other issues (twos complement turns to very large number), which is why I removed it.  In the case the the caller sent NULL for both args1 and args2, count_args would return zeros, then the -1 is problematic.

Probably the correct solution here is to have _count_args return just the correct number of arguments.